functions/src/notifications/deliverNotifications.ts

```diff
@@ -83,22 +85,23 @@ const deliverEmailNotifications = async () => {
     } else {
       const htmlString = renderToHtmlString(digestData)
 
-      // Create an email document in /emails to queue up the send
-      await db.collection("emails").add({
+      const email = {
         to: [verifiedEmail],
         message: {
           subject: "Your Notifications Digest",
           text: "", // blank because we're sending HTML
           html: htmlString
         },
         createdAt: Timestamp.now()
-      })
+      }
+      batch.create(db.collection("emails").doc(), email)
 
-      console.log(`Saved email message to user ${profileDoc.id}`)
+      console.log(`Saving email message to user ${profileDoc.id}`)
     }
 
     const nextDigestAt = getNextDigestAt(profile.notificationFrequency)
-    await profileDoc.ref.update({ nextDigestAt })
+    batch.update(profileDoc.ref, { nextDigestAt })
+    await batch.commit()
```

Comment:
Is there anything to cause an error to be thrown in this scenario? Does getNextDigestAt throw an error? If so, should we wrap the batch.commit() with a try catch to tell us when it does fail?

Reply:
No, but given that these two updates are intrinsically linked, I want to make sure they're part of one transaction - given we're doing all these writes async, I think it's possible for something else in the firebase function to fail in between these steps and bring the whole thing down. Adding a transaction here should mean (at least from a business logic point of view) we can eliminate a class of bug entirely